mlflow/gateway/providers/__init__.py

```diff
@@ -0,0 +1,14 @@
+from mlflow.exceptions import MlflowException
+from .base import BaseProvider
+from .openai import OpenAIProvider
+from ..config import Provider
+
+
+def get_provider(provider: Provider) -> BaseProvider:
+    provider_to_class = {
+        Provider.OPENAI: OpenAIProvider,
+    }
+    if prov := provider_to_class.get(provider):
+        return prov
+
```

Comment:
```suggestion
if prov := find_custom_provider(...):
    return prov
```

We can fetch a custom provider here, but we don't need to do that for now.